src/main/kotlin/com/cognifide/gradle/common/file/transfer/FileTransferManager.kt

```diff
@@ -32,21 +32,21 @@ class FileTransferManager(private val common: CommonExtension) : FileTransfer {
         common.prop.string("fileTransfer.password")?.let { set(it) }
     }
 
+    var bearerToken: String? = null
+
     val domain = common.obj.string {
         common.prop.string("fileTransfer.domain")?.let { set(it) }
     }
 
-    val credentials: Pair<String, String> get() = when {
-        user.isPresent && password.isPresent -> (user.get() to password.get())
-        else -> throw FileTransferException("File transfer credentials are missing!")
-    }
+    val credentials: Pair<String, String>?
+        get() = if (user.isPresent && password.isPresent) user.get() to password.get() else null
 
-    val credentialsString get() = credentials.run { "$first:$second" }
+    val credentialsString get() = credentials?.run { "$first:$second" } ?: ""
```

Comment:
the empty string is not a solution;

the caller need to check for nullability or availability of that property
Kotlin intentionally has nullable types to check defined values at compile-time; using empty strings is a form of hacking that mechanism

Reply:
makes sense, fixed, thanks!